pkg/controller/cronjob/cronjob_controller2.go

```diff
@@ -0,0 +1,580 @@
+		if found := childrenJobs[j.UID]; !found {
+			recorder.Eventf(cj, v1.EventTypeNormal, "MissingJob", "Active job went missing: %v", j.Name)
+			deleteFromActiveList(cj, j.UID)
+		}
+	}
+
+	// TODO: @alpatel explore if cached client can be used as realCJControl
+	updatedCJ, err := cjc.UpdateStatus(cj)
+	if err != nil {
+		klog.V(2).Infof("Unable to update status for %s (rv = %s): %v", nameForLog, cj.ResourceVersion, err)
+		return err, nil
+	}
+	*cj = *updatedCJ
+
+	if cj.DeletionTimestamp != nil {
+		// The CronJob is being deleted.
+		// Don't do anything other than updating status.
+		return fmt.Errorf("cronjob %s/%s is being deleted", cj.Namespace, cj.Name), nil
+	}
+
+	if cj.Spec.Suspend != nil && *cj.Spec.Suspend {
+		klog.V(4).Infof("Not starting job for %s because it is suspended", nameForLog)
+		return nil, nil
+	}
+
+	// TODO: alpatel: this is here now just to pass the unit tests,
+	//  move this to the start of the function to return on error early
+	sched, err := cron.ParseStandard(cj.Spec.Schedule)
+	if err != nil {
+		// this is likely a user error in defining the spec value
+		// we should log the error and not reconcile this cronjob until an update to spec
+		klog.Errorf("unparseable schedule: %s : %s", cj.Spec.Schedule, err)
+		recorder.Eventf(cj, v1.EventTypeWarning, "UnparseableSchedule", "unparseable schedule: %s : %s", cj.Spec.Schedule, err)
+		return nil, nil
+	}
+	times, err := getRecentUnmetScheduleTimes2(*cj, now, sched)
+	switch {
+	case err != nil && len(times) == 0:
+		// too many missed jobs, schedule the next one on time and return
+		// TODO: @alpatel, with revised workflow this probably needs be reworked
+		// 		the thought process is we will always miss the schedule time and
+		// 		controller will reconcile after scheduled time + delta time spent
+		//		in reconciliation loop. With that if a job misses 100 schedule times
+		// 		with this block, it will always return here.
+		recorder.Eventf(cj, v1.EventTypeWarning, "TooManyMissedTimes", "Too many missed times for the cronjob, will schedule the next one", err)
+		klog.Error("Too many missed times for the cronjob, will schedule the next one", nameForLog, err)
+		return nil, nil
+	case err != nil:
+		// return error
+		recorder.Eventf(cj, v1.EventTypeWarning, "FailedNeedsStart", "Cannot determine if job needs to be started: %v", err)
+		klog.Errorf("Cannot determine if %s needs to be started: %v", nameForLog, err)
+		return err, nil
+	case len(times) == 0:
+		// no unmet start time, return.
+		// The only time this should happen is if queue is filled after restart.
+		// Otherwise, the queue is always suppose to trigger sync function at the time o
+		// the scheduled time, that will give atleast 1 unmet time schedule
+		klog.V(4).Infof("No unmet start times for %s", nameForLog)
+		return nil, nil
+	default:
+		// multiple unmet start times, start the last one.
+		klog.V(4).Infof("Multiple unmet start times for %s so only starting last one", nameForLog)
```

Comment:
This won't work as you wish to, since `getRecentUnmetScheduleTimes2` returns error when 100+ missed times. You need to react differently or change that function.